src/cargo/core/compiler/custom_build.rs

```diff
@@ -416,6 +416,7 @@ fn build_work(build_runner: &mut BuildRunner<'_, '_>, unit: &Unit) -> CargoResul
                 let values = cfg_map.entry(k.as_str()).or_default();
                 values.push(v.as_str());
             }
+            Cfg::Version(..) => todo!(),
```

Comment:
No idea what to do here

Reply:
I think the following code will be relevant:
https://github.com/rust-lang/cargo/blob/47c911e9e6f6461f90ce19142031fe16876a3b95/src/cargo/ops/cargo_compile/mod.rs#L495-L511

Reply:
The code you point to wasn't helpful. I still don't know in what format I should write the version cfg as an env var.

Reply:
Just to be clear here, we do not expect rustc to emit this in `--print`, correct? That means this code path should never be reached, right?

And, similar to the point made at https://github.com/rust-lang/rust/pull/141137#discussion_r2096067981, it doesn't really make sense to include this here since it would need to emit *every* version to be correct. My inclination is that the env var shouldn't be set for build scripts.

Reply:
Ah, yes, I missed what this code was doing. I was giving code to help when evaluating dependencies.  I agree with Eric that we shouldn't do anything here.  Maybe worth a `trace!`